### PR TITLE
RSDK-5917 - Increase module startup timeout

### PR DESCRIPTION
**Pull request by cheukt**
adds a separate var (+ separate env var)

oueTimeout has been removed as well

### CODE REVIEW

**Comment by benjirewis** [comment]
> should we up the onUnexpectedExit handler timeout as well?

The OUE handler should be using the same timeouts as regular module startup already, no? It just calls into `startProcess` and `checkReady` in the same way as `add`?

File: utils/errors.go [hunk 1]
```diff
@@ -22,3 +26,30 @@ func TypeStr[T any]() string {
...
+	return timeoutErrorHelper(id, timeout, timeoutMsg, DefaultModuleStartupTimeout, ModuleStartupTimeoutEnvVar)
+}
+
+func timeoutErrorHelper(id string, timeout time.Duration, timeoutMsg string, defaultTimeout time.Duration, timeoutEnvVar string) error {
+	extraMsg := ""
+	if timeout == defaultTimeout {
+		extraMsg = fmt.Sprintf(" Update %s env variable to override", timeoutEnvVar)
+	}
+	return fmt.Errorf(
+		"%s timed out after %v during %v.%v",
```
Comment (benjirewis) [inline utils/errors.go:52]: Is this the same error message as before or a different one? I can't quite tell with all the formatting directives.

**Review by benjirewis — APPROVED** [review]
Nice! Pretty much LGTM, just checking if we're maintaining the same log messages (I realize name of the environment variable will change depending on module vs. resource timeout).

**Comment by cheukt** [comment]
theres a 2 minutes `oueTimeout`, which is for the entire attempted module restart. Feels like having it be 20mins might be a bit too long tho, but maybe at least 5mins

File: utils/errors.go (hunk 1 above)
Comment (benjirewis) [reply utils/errors.go:52]: Yeah that seems better; I was worried people may be relying on the language of the original log message but I'm probably being overly defensive.

File: module/modmanager/manager.go [hunk 2]
```diff
@@ -627,9 +620,9 @@ func (mgr *Manager) newOnUnexpectedExitHandler(mod *module) func(exitCode int) b
 		// and we should remove orphaned resources. Since we handle process
 		// restarting ourselves, return false here so goutils knows not to attempt
 		// a process restart.
-		if orphanedResourceNames := mgr.attemptRestart(ctx, mod); orphanedResourceNames != nil {
+		if orphanedResourceNames := mgr.attemptRestart(mgr.restartCtx, mod); orphanedResourceNames != nil {
```
[the remark this answers is not in the document]
Comment (cheukt) [reply module/modmanager/manager.go:623]: I think it was fine - ctx inherited from mgr.restartCtx

**Pull request merged**